Userland/Libraries/LibWeb/Layout/BlockContainer.cpp

```diff
@@ -26,6 +27,12 @@ Painting::PaintableWithLines const* BlockContainer::paintable_with_lines() const
     return static_cast<Painting::PaintableWithLines const*>(Box::paintable_box());
 }
 
+void BlockContainer::store_layout_inside_run_info(LayoutState* layoutState, AvailableSpace const availableSpace)
+{
+    document().save_layout_state(layoutState);
+    m_run_available_space = const_cast<AvailableSpace&>(availableSpace);
```

Comment:
available space is a value that exists only within layout process and we should not leak it by caching inside layout nodes (you may have noticed that during layout process the same box is laid out many times with a different available spaces, so it's confusing to cache value used for the last invocation). instead it should be possible to determine the available space for a partial layout based on the layout state calculated by the previous layout run.